pkg/cloud/services/eks/securitygroup_test.go

```diff
@@ -0,0 +1,222 @@
+	g.Expect(desiredTags).To(Equal(expectedDesiredTags))
+}
+
+func TestUpdateTagsForEKSManagedSecurityGroup(t *testing.T) {
+	g := NewWithT(t)
+
+	mockControl := gomock.NewController(t)
+	defer mockControl.Finish()
+
+	eksMock := mock_eksiface.NewMockEKSAPI(mockControl)
+	iamMock := mock_iamauth.NewMockIAMAPI(mockControl)
+	ec2Mock := mocks.NewMockEC2API(mockControl)
+
+	scheme := runtime.NewScheme()
+	_ = infrav1.AddToScheme(scheme)
+	_ = ekscontrolplanev1.AddToScheme(scheme)
+	client := fake.NewClientBuilder().WithScheme(scheme).Build()
+	vpcSpec := infrav1.VPCSpec{
+		IPv6: &infrav1.IPv6{
+			CidrBlock: "2001:db8:85a3::/56",
+		},
+	}
+	scope, err := scope.NewManagedControlPlaneScope(scope.ManagedControlPlaneScopeParams{
+		Client: client,
+		Cluster: &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "ns",
+				Name:      "cluster-name",
+			},
+		},
+		ControlPlane: &ekscontrolplanev1.AWSManagedControlPlane{
+			Spec: ekscontrolplanev1.AWSManagedControlPlaneSpec{
+				RoleName: ptr.To[string]("arn-role"),
+				Version:  aws.String("1.29"),
+				Region:   "us-east-1",
+				NetworkSpec: infrav1.NetworkSpec{
+					Subnets: []infrav1.SubnetSpec{
+						{
+							ID:               "sub-1",
+							CidrBlock:        "10.0.10.0/24",
+							AvailabilityZone: "us-west-2a",
+							IsPublic:         true,
+							IsIPv6:           true,
+							IPv6CidrBlock:    "2001:db8:85a3:1::/64",
+						},
+						{
+							ID:               "sub-2",
+							CidrBlock:        "10.0.10.0/24",
+							AvailabilityZone: "us-west-2b",
+							IsPublic:         false,
+							IsIPv6:           true,
+							IPv6CidrBlock:    "2001:db8:85a3:2::/64",
+						},
+					},
+					VPC: vpcSpec,
+				},
+			},
+		},
+	})
+	g.Expect(err).To(BeNil())
+
+	s := NewService(scope)
+	s.EKSClient = eksMock
+	s.IAMClient = iamMock
+	s.EC2Client = ec2Mock
+
+	existingTags := infrav1.Tags{
+		"aws:eks:cluster-name":               "cluster-name",
+		"kubernetes.io/cluster/cluster-name": "owned",
+		"Name":                               "eks-cluster-sg-cluster-name-1488456827",
+	}
+
+	desiredTags := infrav1.Tags{
+		"hello":                              "there",
+		"new":                                "tag",
+		"aws:eks:cluster-name":               "cluster-name",
+		"kubernetes.io/cluster/cluster-name": "owned",
+		"Name":                               "eks-cluster-sg-cluster-name-1488456827",
+		"tagKey":                             "tagValue",
+		"sampleTag":                          "sampleNewValue",
+	}
+
+	sampleEKSecurityGroupID := aws.String("sg-025f2495c64d5")
+
+	createSGinput := &ec2.CreateSecurityGroupInput{
+		GroupName: sampleEKSecurityGroupID,
+		TagSpecifications: []*ec2.TagSpecification{
+			{
+				Tags: converters.MapToTags(desiredTags),
+			},
+		},
+	}
+
+	ec2Mock.EXPECT().CreateSecurityGroup(createSGinput).Return(&ec2.CreateSecurityGroupOutput{
+		GroupId: sampleEKSecurityGroupID,
+	}, nil)
+
+	_, err = ec2Mock.CreateSecurityGroup(createSGinput)
```

Comment:
You cannot call the mock methods yourself in a test. They're supposed to be called by CAPA code.